woocommerce/woocommerce-ios
# Changes OrderItem.quantity from Int to Decimal

Closes #662, #125

Adds a new data model for OrderItem.quantity to move to a Decimal data type to support extensions that change it from Integer. This is useful for stores that sell partial quantities (like by weight in 1.5 pounds).

I added `IterativeStoreMigrator` that is based off of the ALIterativeMigrator used in the WordPress app. This is an entirely new Swift file and probably needs some attention to code style and Swift-ness. I tried to do as good as I could but I'm a tad rusty and could really use your help.

There are two unit tests that cover the iterative migrator. This ensures that Model 1 to Model 10 fails without using the iterative migrator, and then succeeds when it is used.

# Testing
1. Switch back to develop. Clean, rebuild, and launch.
2. Sign in, make sure to fetch orders.
3. Kill the app.
4. Close Xcode, switch to this branch.
5. Reload Xcode, clean, rebuild, and launch.
6. Switch to orders tab - orders should already be present.

Update release notes:
